java/ql/lib/semmle/code/java/dataflow/internal/DataFlowUtil.qll

```diff
@@ -194,12 +194,14 @@ predicate simpleAstFlowStep(Expr e1, Expr e2) {
   // In the following three cases only record patterns need this flow edge, leading from the bound instanceof
   // or switch tested expression to a record pattern that will read its fields. Simple binding patterns are
   // handled via VariableAssign.getSource instead.
+  // We only consider unique patterns because cases that declare multiple patterns are not allowed to declare
+  // any identifiers, so can't participate in dataflow.
```

Comment:
For consistency, would it be better to only step into patterns that declare an identifier, so we don't step into `case X _:` but not `case X _, Y _:`?